src/components/pages/users.js

```diff
@@ -26,38 +31,201 @@ const availableRoles = Object.values(ROLE);
 
 const TABS = {
   internalUsers: 'internal-users',
-  externalUsers: 'external-users'
+  externalUsers: 'external-users',
+  storageUsers: 'storage-users'
 };
 
-function splitInternalAndExternalUsers(allUsers) {
+function createUserSubsets(users, storagePlans) {
+  const storagePlansById = new Map(storagePlans.map(plan => [plan._id, plan]));
+
   const internalUsers = [];
   const externalUsers = [];
-  for (const user of allUsers) {
-    const matches = (/^external\/(.+)$/).exec(user.provider);
-    if (matches) {
-      externalUsers.push({
-        ...user,
-        importSource: matches[1]
-      });
+  const storageUsers = [];
+
+  for (const user of users) {
+    const enrichedUserObject = {
+      ...user,
+      storagePlan: storagePlansById.get(user.storage?.plan) || null,
+      importSource: (/^external\/(.+)$/).exec(user.provider)?.[1] || null
+    };
+
+    if (enrichedUserObject.importSource) {
+      externalUsers.push(enrichedUserObject);
     } else {
-      internalUsers.push(user);
+      internalUsers.push(enrichedUserObject);
     }
-  }
 
-  return { internalUsers, externalUsers };
-}
+    if (user.storage || user.storagePlan) {
+      storageUsers.push(enrichedUserObject);
```

Comment:
Should we maybe decide here based on the occupied storage value instead of the `storagePlan`?
Basically any case in which a user occupies storage space, so nothing falls through the cracks.

Reply:
You are right, but even worse, the second condition makes no sense at all, it can never be true while the first one is false. I changed it to `user.storage?.plan || user.storage?.usedStorageInBytes`, so nothing can fall through the cracks now.

Reply:
Oh right, there is no `user.storagePlan`, there's `user.storage.plan`